Tools/scripts/validate_board_hwdef.py

```diff
@@ -0,0 +1,353 @@
+                text=True,
+                check=True
+            )
+            return result.stdout.splitlines()
+        except subprocess.CalledProcessError:
+            return []
+
+    content_parts = []
+
+    # Read files from main repository
+    for relative_path in get_git_files(repo_root):
+        if not should_check_file(relative_path):
+            continue
+
+        filepath = repo_root / relative_path
+        try:
+            content = filepath.read_text(encoding='utf-8', errors='ignore')
+            content_parts.append(content)
+        except Exception:
+            # Skip files that can't be read
+            pass
+
+    # Also read files from submodules
+    try:
+        result = subprocess.run(
+            ['git', 'submodule', 'foreach', '--quiet', 'echo $sm_path'],
+            cwd=repo_root,
+            capture_output=True,
+            text=True,
+            check=True
+        )
+        submodule_paths = result.stdout.splitlines()
+
+        for submodule_path in submodule_paths:
+            submodule_dir = repo_root / submodule_path
+            if not submodule_dir.exists():
+                continue
+
+            for relative_path in get_git_files(submodule_dir):
+                if not should_check_file(relative_path):
+                    continue
+
+                filepath = submodule_dir / relative_path
+                try:
+                    content = filepath.read_text(encoding='utf-8', errors='ignore')
+                    content_parts.append(content)
+                except Exception:
+                    # Skip files that can't be read
+                    pass
+    except subprocess.CalledProcessError:
+        pass
+
+    # Join all content with newlines and cache it
+    _codebase_content_cache = '\n'.join(content_parts)
+    return _codebase_content_cache
+
+
+def validate_board_worker(args: Tuple) -> Tuple[str, Optional[Dict[str, str]], Optional[str]]:
+    """
+    Worker function for parallel board validation.
+
+    Args:
+        args: Tuple of (board, board_num, total_boards, verbose)
+
+    Returns:
+        Tuple of (board_name, unused_defines_dict, error_message)
+    """
+    board, board_num, total_boards, verbose = args
+
+    # Build codebase content once per worker process (cached)
+    codebase_content = build_codebase_content()
+
+    # Import here to avoid issues with multiprocessing
+    board_list = BoardList()
+
+    try:
+        # Get the HWDef instance for this board
+        hwdef_instance = board_list.get_HWDef_instance(board)
+
+        if hwdef_instance is None:
+            # SITL board - no hwdef file
+            return (board.name, None, None)
+
+        # Process the hwdef files to populate the defines dictionary
+        hwdef_instance.process_hwdefs()
+
+        # Get defines from the hwdef instance
+        defines = hwdef_instance.defines
+
+        # Check each define
+        unused = {}
+        for define_name, line in defines.items():
```

Comment:
```suggestion
        for define_name, value in defines.items():
```

Or should the other one be `other_line`?